src/preferences/dialog/dlgprefrecord.cpp

```diff
@@ -70,6 +80,41 @@ DlgPrefRecord::DlgPrefRecord(QWidget* parent, UserSettingsPointer pConfig)
 
     setupEncoderUI();
 
+    // generate custom recording samplerates
+    const QList<mixxx::audio::SampleRate>* customSampleRatesList = nullptr;
+    if (prefformat == ENCODING_OPUS) {
+        customSampleRatesList = &kRecSampleRatesOpus;
+    } else if (prefformat == ENCODING_MP3) {
+        customSampleRatesList = &kRecSampleRatesMP3;
+    } else if (prefformat == ENCODING_OGG) {
+        customSampleRatesList = &kRecSampleRatesOGG;
+    } else {
+        customSampleRatesList = &kRecSampleRates;
+    }
+    updateSampleRates(*customSampleRatesList);
```

Comment:
The pointer type can be replaced here with a reference, to avoid the nulptr case, but results to the same machine code:

```suggestion
    const QList<mixxx::audio::SampleRate>& customSampleRatesList = kRecSampleRates;
    if (prefformat == ENCODING_OPUS) {
        customSampleRatesList = kRecSampleRatesOpus;
    } else if (prefformat == ENCODING_MP3) {
        customSampleRatesList = kRecSampleRatesMP3;
    } else if (prefformat == ENCODING_OGG) {
        customSampleRatesList = kRecSampleRatesOGG;
    } 
    updateSampleRates(customSampleRatesList);
```